src/core/infrastructure/event.spec.js

```diff
@@ -1,4 +1,39 @@
-describe('event', () => {
+import { Event } from './event';
 
+describe('Event', () => {
+	let event;
+
+	beforeEach(() => {
+		event = new Event();
+		window.testValue = 0;
+	});
+
+	let foo = function (value, func) {
+		func();
+		++window.testValue;
+	};
+
+	describe('on/emit', () => {
+
+		it('should return 1 in both cases', () => {
```

Comment:
Consider to use [Chai spies](http://chaijs.com/plugins/chai-spies/) .
In this case you won't need to test side effects in window object. It'll be look like this:
```javascript
const foo = chai.spy();
event.on(foo);
expect(foo).to.have.been.called();
```